src/main/java/uk/gov/hmcts/reform/unspec/handler/callback/AcknowledgeServiceCallbackHandler.java

```diff
@@ -66,17 +68,17 @@ private CallbackResponse validateDateOfBirth(CallbackParams callbackParams) {
     }
 
     private CallbackResponse setNewResponseDeadline(CallbackParams callbackParams) {
-        Map<String, Object> data = callbackParams.getRequest().getCaseDetails().getData();
-        LocalDateTime responseDeadline = callbackParams.getCaseData().getRespondentSolicitor1ResponseDeadline();
-
+        CaseData caseData = callbackParams.getCaseData();
+        LocalDateTime responseDeadline = caseData.getRespondentSolicitor1ResponseDeadline();
         LocalDate newResponseDate = workingDayIndicator.getNextWorkingDay(responseDeadline.plusDays(14).toLocalDate());
+        CaseData.CaseDataBuilder caseDataBuilder = caseData.toBuilder()
+            .respondentSolicitor1ResponseDeadline(newResponseDate.atTime(MID_NIGHT));
 
-        data.put(RESPONSE_DEADLINE, newResponseDate.atTime(MID_NIGHT));
-        List<String> errors = businessProcessService.updateBusinessProcess(data, ACKNOWLEDGE_SERVICE);
+        CaseData caseDataUpdated = businessProcessService.updateBusinessProcess(
```

Comment:
updateBusinessProcess returning CaseData seems a lot? I would have thought this would be nicer as a void method on BusinessProcess? I'm not actually sure of a use for BusinessProcessService. Again maybe not this PRs concern

Reply:
I agree that `BusinessProcessService` seems unnecessary and we can clean that up as separate PR not part of this.